ros2/rust/string_length_node/src/main.rs

```diff
@@ -0,0 +1,69 @@
+// Rust imports
+use std::sync::{Arc, Mutex};
+// rclrs related imports
+use rclrs::{Publisher, RclrsError};
+use std_msgs::msg::String as StringMsg;
+use std_msgs::msg::UInt8 as UInt8Msg;
+
+// Define the struct for the node
+struct StringLengthNode {
+    node: Arc<rclrs::Node>,
+    _subscription: Arc<rclrs::Subscription<StringMsg>>,
+    data: Arc<Mutex<Option<StringMsg>>>,
+    publisher: Arc<rclrs::Publisher<UInt8Msg>>,
+}
+
+// Define the implementation of the node.
+impl StringLengthNode {
+    // This function is called when creating the node.
+    fn new(context: &rclrs::Context) -> Result<Self, rclrs::RclrsError> {
+        let node = rclrs::Node::new(context, "string_length_node")?;
+        let data = Arc::new(Mutex::new(None));
+        let data_cb = Arc::clone(&data);
```

Comment:
You can use `data.clone()` instead of `Arc::clone`. Some people prefer the later, but I find it harder to read.

They do the same thing: https://doc.rust-lang.org/std/sync/struct.Arc.html#cloning-references